pkg/devicemapper/devmapper.go

```diff
@@ -264,14 +259,6 @@ func UdevWait(cookie *uint) error {
 	return nil
 }
 
-var dmLogger DevmapperLogger
-
-// LogInit initializes the logger for the device mapper library.
-func LogInit(logger DevmapperLogger) {
-	dmLogger = logger
-	LogWithErrnoInit()
-}
```

Comment:
The semantics of this function never made sense because you **need** to have callbacks registered for this package to function. Which meant that a user of this package has to explicitly call this function so that the package will work as intended. That doesn't seem like a sane interface to me.

If you like, I can re-add `LogInit` but change it's semantics so that it acts as an additional logger (or a replacement for the `logrus` one we have now). But this will also "break" users in a more drastic way (rather than not compiling, code that worked previously will have slightly different semantics).

As for breaking people, surely everyone uses `vendor` these days so I'm not really sure what you mean by this. Yes, when they update they will have to make **very minor** changes but the alternative is to keep an old interface that will lead people to think that the semantics are unchanged (or that the semantics are different to what they actually are).